src/screens/Home/Home.js

```diff
@@ -558,6 +553,11 @@ class HomeScreen extends React.Component<Props, State> {
             contentContainerStyle={{ flexGrow: 1 }}
           />
         </ScrollView>
+        <ReferralModalReward
+          isVisible={showRewardModal}
+          onModalHide={(callback) => this.setState({ showRewardModal: false },
```

Comment:
What do you think if just move this navigation `callback` from component level to screen and avoid that `callback` in general?

Reply:
Initially I did so,  but then figured that it's better not to add state to Modal component and keep it on parent component. 
And by adding state to ReferralModalReward, additional methods are needed to track isVisible prop changes. So I thought callback solves modal closing in cleaner way.

If those arguments are not that sound, I can adjust this part :)